File: 计算机/算法/Algorithm/Java/zuo/TreeErgodic.java
package Java.zuo;
//递归遍历树
public class TreeErgodic {
    public static void main(String[] args) {
        TreeNode1 treeNode1 = new TreeNode1();
        TreeNode1 treeNodeleft11 = new TreeNode1();
        TreeNode1 treeNoderight11 = new TreeNode1();
        TreeNode1 treeNodeleft21 = new TreeNode1();
        TreeNode1 treeNoderight21 = new TreeNode1();
        TreeNode1 treeNodeleft31 = new TreeNode1();
        TreeNode1 treeNoderight31 = new TreeNode1();

        treeNode1.value = 1;
        treeNodeleft11.value = 2;
        treeNoderight11.value = 3;
        treeNodeleft21.value = 4;
        treeNoderight21.value = 5;
        treeNodeleft31.value = 6;
        treeNoderight31.value = 7;
        treeNode1.left = treeNodeleft11;
        treeNode1.right = treeNoderight11;
        treeNodeleft11.left = treeNodeleft21;
        treeNodeleft11.right = treeNoderight21;
        treeNoderight11.left = treeNodeleft31;
        treeNoderight11.right = treeNoderight31;
        System.out.println("---先序遍历---");
        preOrder(treeNode1);
        System.out.println("---中序遍历---");
        inOrder(treeNode1);
        System.out.println("---后序遍历---");
        posOrder(treeNode1);

    }

    // 先序遍历
    public static void preOrder(TreeNode1 head) {
        if (head == null) {
            return;
        }
        System.out.println(head.value);
        preOrder(head.left);
        preOrder(head.right);

    }

    // 中序遍历
    public static void inOrder(TreeNode1 head) {
        if (head == null) {
            return;
        }
        inOrder(head.left);
        System.out.println(head.value);
        inOrder(head.right);
    }

    // 后序遍历
    public static void posOrder(TreeNode1 head) {
        if (head == null) {
            return;
        }
        posOrder(head.left);
        posOrder(head.right);
        System.out.println(head.value);
    }

}

// 树结构

class TreeNode {
    TreeNode1 left;
    TreeNode1 right;
    int value;

}

File: 算法/Algorithm/Go/zuo/TreeErgodic.go
package main

import "fmt"

type TreeNode struct {
	Value int
	Left  *TreeNode
	Right *TreeNode
}

//递归遍历树
func main() {
	treeNode := &TreeNode{Value: 1}
	treeNodeLeft1 := &TreeNode{Value: 2}
	treeNodeRight1 := &TreeNode{Value: 3}
	treeNodeLeft2 := &TreeNode{Value: 4}
	treeNodeRight2 := &TreeNode{Value: 5}
	treeNodeLeft3 := &TreeNode{Value: 6}
	treeNodeRight3 := &TreeNode{Value: 7}

	treeNode.Left = treeNodeLeft1
	treeNode.Right = treeNodeRight1
	treeNodeLeft1.Left = treeNodeLeft2
	treeNodeLeft1.Right = treeNodeRight2
	treeNodeRight1.Left = treeNodeLeft3
	treeNodeRight1.Right = treeNodeRight3

	fmt.Println("---先序遍历---")
	preOrder(treeNode)
	fmt.Println("---中序遍历---")
	inOrder(treeNode)
	fmt.Println("---后序遍历---")
	posOrder(treeNode)
}

// 先序遍历
func preOrder(head *TreeNode) {
	if head == nil {
		return
	}
	fmt.Println(head.Value)
	preOrder(head.Left)
	preOrder(head.Right)
}

// 中序遍历
func inOrder(head *TreeNode) {
	if head == nil {
		return
	}

	inOrder(head.Left)
	fmt.Println(head.Value)
	inOrder(head.Right)
}

// 后序遍历
func posOrder(head *TreeNode) {
	if head == nil {
		return
	}

	posOrder(head.Left)
	posOrder(head.Right)
	fmt.Println(head.Value)
}
